In [121]:
import os 
import re
import calendar
import numpy as np

In [122]:
path = "./data/"

In [123]:
# arrays para guardar datos
days = np.array([], dtype=int)
months = np.array([], dtype=int)
years = np.array([], dtype=int)

In [124]:
def read_txt_files(path, days, months, years):
  with open(path, 'r', encoding='utf-8-sig') as f:
    dates = [line.rstrip() for line in f]
    
    # Los paréntesis en una expresión regular delimitan un grupo de captura
  day_regex =  r'^(0?[1-9]|[1-2][0-9]|3[0-1])(\.|-|/)'
  month_regex =  r'(\.|-|/)(0?[1-9]|1[0-2]|[A-Za-z]{3,})(\.|-|/)'
  year_regex =  r'(\.|-|/)(20([0-1][0-9]))$'
  
  for date in dates:
    day_match  = re.search(day_regex, date)
    day = day_match.group(1)
    days = np.append(days, day)
    
    month_match = re.search(month_regex, date)
    month = month_match.group(2)
    
    # Verificar si la fecha es un str de numeros, si no lo es, uso calendar
    # para obtener su valor
    if month.isdigit():
      month_num  = month
    else:
      month_num = list(calendar.month_abbr).index(month.capitalize())
      month_num  = f"{month_num:02}"
                
    months = np.append(months, month_num)
    
    year_match = re.search(year_regex, date)
    year = year_match.group(2)
    years = np.append(years, year)
      
  return days, months, years
    

In [125]:
for file in os.listdir(path): 
  if file.endswith(".txt"): 
    file_path = f"{path}\{file}"
    days, months, years = read_txt_files(file_path, days, months, years)

In [130]:
days = days.astype(int)
days.size

21000

In [131]:
months = months.astype(int)
months.size

21000

In [132]:
years = years.astype(int)
years.size

21000

In [133]:
print("El promedio de días es: {:.2f}, de meses: {:.2f}, de años: {:.2f}".format(np.mean(days), np.mean(months), np.mean(years)))


El promedio de días es: 15.62, de meses: 6.47, de años: 2016.69


### **Respuesta**
---
El promedio de días es: 15.62, de meses: 6.47, de años: 2016.69